app/controllers/application_controller.rb

```diff
@@ -8,7 +8,10 @@ class ApplicationController < ActionController::Base
   def current_user
     # Overwrites the session cookie if an extended_session cookie exists
     session[:user_id] ||= cookies.encrypted[:_extended_session]['user_id'] if cookies.encrypted[:_extended_session].present?
-    @current_user ||= User.find_by(id: session[:user_id])
+
+    user = User.find_by(id: session[:user_id])
+    user = nil unless valid_session?(user)
+    @current_user ||= user
```

Comment:
There's a small possibility where a request that should have its session ended will circumvent the check,
The `@current_user` is memoizing the first none `nil` value so **L13** will have only an effect on fresh requests.

Reply:
Besides, catching an invalid session should invalidate (reset) it to reduce error surface.